src/foam/dao/RoutingJournal.js

```diff
@@ -9,9 +9,25 @@ foam.CLASS({
   name: 'RoutingJournal',
   extends: 'foam.dao.FileJournal',
 
-  documentation:
-    `Journal interface that also adds the DAO name to the journal entry so that one may use
-    a single journal file and still be able to put the entry into the correct DAO`,
+  documentation: `Journal interface that also adds the DAO name to the journal
+    entry so that one may use a single journal file and still be able to put the
+    entry into the correct DAO`,
+
+  javaImports: [
+    'foam.dao.java.FindReplayDAO'
+  ],
+
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        cls.extras.push(`
+          // Boolean to check if the journal file is being or has been replayed
+          protected volatile boolean journalReplayed_ = false;
```

Comment:
I think this would also not be needed if you had a PromisedDAO.

Reply:
No, I think I will need this. Wnen the first service initiates a `replay`, all of the records in the journal will be attempted to be read into the DAOs. Other DAOs would also initiate a replay as the PromisedDAO is a decorator around individual DAOs.